docs/api.md

```diff
@@ -0,0 +1,158 @@
+# API
+
+The recommendation server provides a single API endpoint that, givene a partial search string, will attempt to complete that search and make a destination recommendation.
+
+
+## Request
+
+GET requests should be made to the root of the server with a single querystring parameter containing the query:
+
+```bash
+GET http://universal-search.dev/?q=the%20mart
+```
+
+## Response Status
+
+The HTTP status code will tell you about the status of the request:
+
+* `200` - request successfully received, successfully returned recommendation. The response body will contain the recommendation.
+* `202` - request successfully received, but the recommendation was not contained in cache. The response body will contain an empty object literal.
+* `400` - the request did not contain a query.
+* `500` - application error. If running in `DEBUG` mode, details about the exception will be contained in the response body.
+
+
+## Response Body
+
+If the status code indicates that a recommendation has been returned, the response body will contain that recommendation following this general form:
+
+```json
+{
+  "enhancers": {
+    "wikipedia": {"...": "..."},
+    "omdb": {"...": "..."}
+  },
+  "query": {
+    "completed": "the martian",
+    "original": "the mart"
+  },
+  "result": {
+    "abstract": "Directed by Ridley Scott. With Matt Damon, Jessica Chastain, Kristen Wiig, Kate Mara. During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a...",
+    "title": "The Martian (2015) - IMDb",
+    "url": "http://www.imdb.com/title/tt3659388/"
+  }
+}
+```
+
+There will always be three keys in the root object:
+
+* `enhancers` will be an object containing [context-specific metadata](#enhancers) for the destination recommendation provided in `result`.
```

Comment:
Maybe mention that `enhancers` is always present but may be empty if no enhancers were applied